### PR TITLE
feat: define git blame

### LINKED ISSUE #679: Define GitBlame
## Context
In order to better assign reviewers to a pull request we need to have a clear vision of the list of the files' authors.

The idea of this issue is to be able to understand how to extract the information about the authors of a file line by line, similar to what [git blame](https://www.git-scm.com/docs/git-blame) does.

## Proposed Solution
Define a function following the syntax:
`GitBlame(commitSHA: string, filesPath []string) GitBlame`

### Parameters
- `commitSHA` - the commit from which the specified file should be queried.
- `filesPath` - the list of paths to the file to be queried.

### Return value
The author for each line of the specified `filePath` on the given `commitSHA`. It should follow a struct called `GitBlame` and be defined as follows:
```go
type GitBlameFileLines struct {
	FromLine int
	ToLine   int
	Author   string
}

type GitBlameFile struct {
	FilePath  string
	LineCount int
	Lines     []GitBlameFileLines
}

type GitBlame struct {
	CommitSHA string
	Files     map[string]GitBlameFile // map: filePath => GitBlameFile
}
```

### PR DESCRIPTION
**Pull request by zolamk**
## Description
Defines a `GitBlame` struct and adds method to github client to get git blame information for a set of files.

## Related issue

Closes #679 

## Type of change

New feature (non-breaking change which adds functionality)

## How was this tested?
Manual and unit tests.

## Checklist

- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works 
- [x] I have ran `task check -f` and have no issues

## Code review and merge strategy (ship/show/ask) 

 
 
- [x] Ask: this pull request requires a code review before merge

### CODE REVIEW

File: codehost/github/blame.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+							Login string
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+func (c *GithubClient) GetGitBlame(ctx context.Context, owner, name, commitSHA string, filePaths []string) (*GitBlame, error) {
```
Comment (ferreiratiago) [inline codehost/github/blame.go:50]: Overall the function looks good. We need to perform some stress tests but we will leave that to later on. For now, and following my understanding I believe we can break down this function into small pieces. Here's my suggestion:

```go
func (c *GithubClient) GetGitBlame(ctx context.Context, owner, name, commitSHA string, filePaths []string) (*GitBlame, error) {
	graghQLQuery := builtGraphQLQueryGitBlame(...)

	// ...

	return mapGitBlameQueryToGitBlame(....)
}
```

WDYT?

Comment (zolamk) [reply codehost/github/blame.go:50]: Yeah, makes sense to break it down

File: codehost/github/blame.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	query, pathToKeyMap := buildGitBlameGraphQLQuery(filePaths)
```
Comment (ferreiratiago) [inline codehost/github/blame.go:51]:
```suggestion
	graphQLSuery, mapQueryKeyToFilePath := buildGitBlameGraphQLQuery(filePaths)
```

File: codehost/github/blame.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+	gitBlameQueryData := map[string]interface{}{
```
Comment (ferreiratiago) [inline codehost/github/blame.go:52]:
```suggestion
	gitBlameQueryVariables := map[string]interface{}{
```

File: codehost/github/blame.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+		"owner":            owner,
+		"name":             name,
+		"objectExpression": commitSHA,
+	}
+
+	rawRes, err := c.GetRawClientGraphQL().ExecRaw(ctx, query, gitBlameQueryData)
```
Comment (ferreiratiago) [inline codehost/github/blame.go:58]:
```suggestion
	rawResponse, err := c.GetRawClientGraphQL().ExecRaw(ctx, query, gitBlameQueryData)
```

File: codehost/github/blame.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+	if len(gitBlameQuery.Repository.Object) == 0 {
+		return nil, fmt.Errorf("error getting blame information: no blame information found")
+	}
+
+	return mapGitBlameQueryToGitBlame(commitSHA, gitBlameQuery, pathToKeyMap)
+}
+
+func buildGitBlameGraphQLQuery(filePaths []string) (string, map[string]string) {
+	query := `query($owner: String!, $name: String!, $objectExpression: String!) {
```
Comment (ferreiratiago) [inline codehost/github/blame.go:77]:
```suggestion
	graphQLQuery := `query($owner: String!, $name: String!, $objectExpression: String!) {
```

File: codehost/github/blame.go [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+				... on Commit {
+					%s
+				}
+			}
+		}
+	}`
+
+	blameQuery := strings.Builder{}
+
+	pathToKeyMap := map[string]string{}
```
Comment (ferreiratiago) [inline codehost/github/blame.go:89]:
```suggestion
	mapQueryKeyToFilePath := map[string]string{}
```

File: codehost/github/blame.go [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+	query = fmt.Sprintf(query, blameQuery.String())
+
+	return query, pathToKeyMap
+}
+
+func mapGitBlameQueryToGitBlame(commitSHA string, gitBlameQuery GitBlameQuery, pathToKeyMap map[string]string) (*GitBlame, error) {
+	files := map[string]GitBlameFile{}
+
+	for key, blame := range gitBlameQuery.Repository.Object {
```
Comment (ferreiratiago) [inline codehost/github/blame.go:120]:
```suggestion
	for queryKey, fileGitBlame := range gitBlameQuery.Repository.Object {
```

File: codehost/github/blame.go [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+		path := pathToKeyMap[key]
```
Comment (ferreiratiago) [inline codehost/github/blame.go:121]:
```suggestion
		filePath := pathToKeyMap[key]
```

**Review by ferreiratiago — COMMENTED** [review]
Some suggestions about the naming. Please have a look :)

**Pull request merged**